### PR TITLE
AgentCore uninstallation and plugin uninstallation

### PR DESCRIPTION
**Pull request by luanwenfei-venus**
【Fix issue】#1236
[Modification content] 1. Agent uninstallation logic 2. Plug-in uninstallation logic 3. TimeUtil in the heartbeat service has been deleted, which is useless and occupied by extra threads
[Use case description] 1. Test cases have been added
[Self-test situation] 1. The local static check has an Exception that the framework needs to be replenished and cannot pass; 2. UT is subsequently merged.
[Scope of impact] 1. Is there any impact on users' use (such as API modifications and startup parameter modifications)? 2. Is it necessary to update some documents (such as user manuals)

### CODE REVIEW

File: sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/AgentCoreEntrance.java [hunk 1]
```diff
@@ -53,6 +56,16 @@ public class AgentCoreEntrance {
      */
     private static int agentType = AgentType.PREMAIN.getValue();
 
+    /**
+     * 缓存当前Agent的产品名
+     */
+    private static String artifactCache;
```
Comment (lilai23) [inline sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/AgentCoreEntrance.java:62]: currentArtifact

File: sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/AgentCoreEntrance.java [hunk 2]
```diff
@@ -53,6 +56,16 @@ public class AgentCoreEntrance {
...
+
+    /**
+     * 缓存当前Agent的adviser
+     */
+    private static AdviserInterface adviserCache;
```
Comment (lilai23) [inline sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/AgentCoreEntrance.java:67]: currentAdviser

File: sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/event/EventManager.java [hunk 3]
```diff
@@ -57,11 +54,15 @@ private EventManager() {
      * 初始化，创建定时任务，定时上报事件
      */
     public static void init() {
-        if (!EVENT_CONFIG.isEnable()) {
+        EventConfig eventConfig = ConfigManager.getConfig(EventConfig.class);
+        if (!eventConfig.isEnable()) {
             LOGGER.info("Event is not enable.");
```
Comment (lilai23) [inline sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/event/EventManager.java]: enabled

File: sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/plugin/PluginManager.java [hunk 4]
```diff
@@ -68,33 +70,71 @@ private PluginManager() {
...
      *
-     * @param plugins 插件名集合
+     * @param pluginNames 插件名集合
      */
-    public static void unInstall(Set<String> plugins) {
-        if (AgentCoreEntrance.isPremain()) {
-            LOGGER.log(Level.WARNING, "Plugins are not allowed to be uninstalled when booting through premain.");
+    public static void uninstall(Set<String> pluginNames) {
+        if (pluginNames == null || pluginNames.isEmpty()) {
+            LOGGER.log(Level.WARNING, "Non plugin is configured to be uninstall.");
```
Comment (lilai23) [inline sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/plugin/PluginManager.java]: no

File: sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/plugin/PluginManager.java [hunk 5]
```diff
@@ -68,33 +70,71 @@ private PluginManager() {
...
      * 卸载插件
      *
-     * @param plugins 插件名集合
+     * @param pluginNames 插件名集合
      */
-    public static void unInstall(Set<String> plugins) {
-        if (AgentCoreEntrance.isPremain()) {
-            LOGGER.log(Level.WARNING, "Plugins are not allowed to be uninstalled when booting through premain.");
+    public static void uninstall(Set<String> pluginNames) {
+        if (pluginNames == null || pluginNames.isEmpty()) {
```
Comment (lilai23) [inline sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/plugin/PluginManager.java]: CollectionUtils.isEmpty可以直接用

File: sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/plugin/PluginManager.java [hunk 6]
```diff
@@ -68,33 +70,71 @@ private PluginManager() {
...
     }
 
     /**
      * 卸载全部插件
      */
-    public static void unInstallAll() {
-
+    public static void uninstallAll() {
+        Set<String> pluginNames = PLUGIN_MAP.keySet();
+        uninstall(pluginNames);
```
Comment (lilai23) [inline sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/plugin/PluginManager.java]: 紧凑点，改成一行

File: sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/config/ConfigManager.java [hunk 7]
```diff
@@ -163,7 +180,7 @@ public void accept(BaseConfig config) {
      * @param classLoader 用于查找加载策略的类加载器，允许在类加载中添加新的配置加载策略
      * @return 加载配置策略
      */
-    private static LoadConfigStrategy<?> getLoadConfigStrategy(File configFile, ClassLoader classLoader) {
+    protected static LoadConfigStrategy<?> getLoadConfigStrategy(File configFile, ClassLoader classLoader) {
```
Comment (lilai23) [inline sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/config/ConfigManager.java]: 包内其他地方未使用，不用改成protected

File: sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/plugin/PluginManager.java (hunk 6 above)
Comment (luanwenfei-venus) [reply sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/plugin/PluginManager.java]: 已修改

File: sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/plugin/PluginManager.java (hunk 4 above)
Comment (luanwenfei-venus) [reply sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/plugin/PluginManager.java]: 已修改

File: sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/plugin/PluginManager.java (hunk 5 above)
Comment (luanwenfei-venus) [reply sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/plugin/PluginManager.java]: 已修改

File: sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/AgentCoreEntrance.java (hunk 1 above)
Comment (luanwenfei-venus) [reply sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/AgentCoreEntrance.java:62]: 没有迭代效果，用于缓存，不修改

File: sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/AgentCoreEntrance.java (hunk 2 above)
Comment (luanwenfei-venus) [reply sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/AgentCoreEntrance.java:67]: 没有迭代效果，用于缓存，不修改

**Review by TangLeDaily — DISMISSED** [review]

File: sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/config/ConfigManager.java (hunk 7 above)
Comment (luanwenfei-venus) [reply sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/config/ConfigManager.java]: 已修改

File: sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/event/EventManager.java (hunk 3 above)
Comment (luanwenfei-venus) [reply sermant-agentcore/sermant-agentcore-core/src/main/java/com/huaweicloud/sermant/core/event/EventManager.java]: 已修改

**Review by TangLeDaily — APPROVED** [review]

**Review by lilai23 — APPROVED** [review]

**Pull request merged**